cpu/stm32/periph/i2c_2.c

```diff
@@ -155,19 +202,19 @@ static void _init(i2c_t dev)
     uint32_t ccr = 0;
     /* read speed configuration */
     switch (i2c_config[dev].speed) {
-        case I2C_SPEED_LOW:
-            /* 10Kbit/s */
-            ccr = i2c_config[dev].clk / 20000;
-            break;
-
-        case I2C_SPEED_NORMAL:
-            /* 100Kbit/s */
-            ccr = i2c_config[dev].clk / 200000;
-            break;
-
-        case I2C_SPEED_FAST:
-            ccr = i2c_config[dev].clk / 800000;
-            break;
+    case I2C_SPEED_LOW:
+        /* 10Kbit/s */
+        ccr = i2c_config[dev].clk / 20000;
```

Comment:
If `i2c_config[dev].speed` would just be the speed in Hz you could do

```suggestion
        ccr = i2c_config[dev].clk / (2 * i2c_config[dev].speed);
```

for arbitrary I2C frequencies. (but this is unrelated to this PR)

Reply:
I added this as a separate PR. The change is IMO trivial enough to not require a separate review.